test/transformations_test.jl

```diff
@@ -15,6 +15,23 @@ include("helper/plain_logic_circuits.jl")
     end
 end
 
+@testset "Structured smooth test" begin
+    sdd = zoo_sdd("random.sdd")
+    vtr = zoo_vtree("random.vtree")
+    slc = smooth(sdd)
+    plc = propagate_constants(sdd, remove_unary=true)
+    structplc = compile(StructLogicCircuit, vtr, plc) 
+    sstructplc = smooth(structplc)
```

Comment:
Why do we need to smooth it twice? Is it because propagage constants need smooth circuit?

Reply:
The smoothed logic circuit is totally orthogonal, it's just for testing. I'm using probabilistic equivalence checking to make sure the smoothing didn't somehow change the meaning of the circuit (that you get the same thing regardless of how you smooth).

Reply:
oh, makes sense, I see now, initially read line 22 wrong.